app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventInitial/EventInitialRepositoryImpl.java

```diff
@@ -96,21 +95,11 @@
     @Override
     public Observable<CategoryCombo> catCombo(String programUid) {
         return d2.programModule().programs().uid(programUid).get()
-                .flatMap(program ->
-                        d2.categoryModule().categoryCombos().withCategories().withCategoryOptionCombos().uid(program.categoryComboUid()).get())
-                .map(categoryCombo -> {
-                    List<Category> fullCategories = new ArrayList<>();
-                    List<CategoryOptionCombo> fullOptionCombos = new ArrayList<>();
-                    for (Category category : categoryCombo.categories()) {
-                        fullCategories.add(d2.categoryModule().categories().withCategoryOptions().uid(category.uid()).blockingGet());
-                    }
-                    List<CategoryOptionCombo> catOptionCombos = d2.categoryModule().categoryOptionCombos()
-                            .byCategoryComboUid().eq(categoryCombo.uid())
-                            .blockingGet();
-                    for (CategoryOptionCombo categoryOptionCombo : catOptionCombos)
-                        fullOptionCombos.add(d2.categoryModule().categoryOptionCombos().withCategoryOptions().uid(categoryOptionCombo.uid()).blockingGet());
-                    return categoryCombo.toBuilder().categories(fullCategories).categoryOptionCombos(fullOptionCombos).build();
-                }).toObservable();
+                .flatMap(program -> d2.categoryModule().categoryCombos()
+                        .withCategories()
+                        .withCategoryOptionCombos()
+                        .uid(program.categoryComboUid()
+                        ).get()).toObservable();
```

Comment:
nice, just make sure if this need to be tested (add info to the ticket pls) thanks